examples/ethereum-tracker/src/contract.rs

```diff
@@ -64,19 +64,25 @@ impl Contract for EthereumTrackerContract {
 }
 
 impl EthereumTrackerContract {
-    fn get_endpoints(&self) -> (EthereumClient, String) {
+    fn get_endpoints(&self) -> (EthereumClient, String, u64) {
         let argument = self.state.argument.get();
         let url = argument.ethereum_endpoint.clone();
         let contract_address = argument.contract_address.clone();
+        let start_block = argument.start_block;
         let ethereum_client = EthereumClient { url };
-        (ethereum_client, contract_address)
+        (ethereum_client, contract_address, start_block)
     }
 
     async fn read_initial(&mut self) {
         let event_name_expanded = "Initial(address,uint256)";
-        let (ethereum_client, contract_address) = self.get_endpoints();
+        let (ethereum_client, contract_address, start_block) = self.get_endpoints();
         let events = ethereum_client
-            .read_events(&contract_address, event_name_expanded, 0)
+            .read_events(
+                &contract_address,
+                event_name_expanded,
+                start_block,
+                start_block + 1,
```

Comment:
Shouldn't it rather read the events from `0` to `start_block`? Otherwise we miss all events before the configured `start_block`, and have wrong balances.

Reply:
No. I mean we could, but that would be a different design.
The events being read in the `fn read_initial` is the initial event that is emitted when the smart contract is initialized.
The other `read_events` are the ones being used when the state is updated in the smart contract.

Reply:
Ah, I see!
Then maybe:
"The creation block" → "The block height at which the EVM contract was created"?

Reply:
Ok, done.